src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -35,6 +35,39 @@
 
 namespace OpenRAVE {
 
+static int64_t ConvertIsoFormatDateTimeToLinuxTimeUS(const char* pIsoFormatDateTime)
+{
+    // RFC 3339 Nano format (2006-01-02T15:04:05.999999999Z07:00)
+    struct tm datetime;
+    const char *remain = strptime(pIsoFormatDateTime, "%FT%T", &datetime);
```

Comment:
please handle the case where `remain` is NULL.

```
RETURN VALUE
       The return value of the function is a pointer to the first character not processed in this function call.  In case the input string contains more  characters  than  required  by  the  format
       string, the return value points right after the last consumed input character.  In case the whole input string is consumed, the return value points to the null byte at the end of the string.
       If strptime() fails to match all of the format string and therefore an error occurred, the function returns NULL.
```

Reply:
@superfashi also handle when `pIsoFormatDateTime=""` and `pIsoFormatDateTime=nullptr`, ie 0 should return

Reply:
done